packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin.swift

```diff
@@ -0,0 +1,422 @@
+      name: "plugins.flutter.io/in_app_purchase",
+      binaryMessenger: messenger)
+    let instance = InAppPurchasePlugin(registrar: registrar)
+    registrar.addMethodCallDelegate(instance, channel: channel)
+    registrar.addApplicationDelegate(instance)
+    SetUpInAppPurchaseAPI(messenger, instance)
+  }
+
+  @objc
+  public init(receiptManager: FIAPReceiptManager) {
+    self.receiptManager = receiptManager
+    self.requestHandlers = Set<FIAPRequestHandler>()
+    self.handlerFactory = {
+      FIAPRequestHandler(request: $0)
+    }
+    super.init()
+  }
+
+  @objc
+  // TODO(louisehsu): this initializer is for tests only. Make this private after the tests are moved to swift.
+  public convenience init(
+    receiptManager: FIAPReceiptManager,
+    handlerFactory: @escaping (SKRequest) -> FIAPRequestHandler
+  ) {
+    self.init(receiptManager: receiptManager)
+    self.handlerFactory = handlerFactory
+  }
+
+  // This is the designated initializer
+  public convenience init(registrar: FlutterPluginRegistrar) {
+    self.init(receiptManager: FIAPReceiptManager())
+    self.registrar = registrar
+
+    self.paymentQueueHandler = FIAPaymentQueueHandler(
+      queue: SKPaymentQueue.default(),
+      transactionsUpdated: { [weak self] transactions in
+        self?.handleTransactionsUpdated(transactions)
+      },
+      transactionRemoved: { [weak self] transactions in
+        self?.handleTransactionsRemoved(transactions)
+      },
+      restoreTransactionFailed: { [weak self] error in
+        self?.handleTransactionRestoreFailed(error as NSError)
+      },
+      restoreCompletedTransactionsFinished: { [weak self] () -> Void in
+        self?.restoreCompletedTransactionsFinished()
+      },
+      shouldAddStorePayment: { [weak self] (payment: SKPayment, product: SKProduct) -> Bool in
+        return self?.shouldAddStorePayment(payment: payment, product: product) ?? false
+      },
+      updatedDownloads: { [weak self] _ in
+        self?.updatedDownloads()
+      },
+      transactionCache: FIATransactionCache())
+    #if os(iOS)
+      let messenger = registrar.messenger()
+    #endif
+    #if os(macOS)
+      let messenger = registrar.messenger
+    #endif
+    transactionObserverCallbackChannel = FlutterMethodChannel(
+      name: "plugins.flutter.io/in_app_purchase", binaryMessenger: messenger)
+  }
+
+  public func canMakePaymentsWithError(_ error: AutoreleasingUnsafeMutablePointer<FlutterError?>)
+    -> NSNumber?
+  {
+    return SKPaymentQueue.canMakePayments() as NSNumber
+  }
+
+  public func transactionsWithError(_ error: AutoreleasingUnsafeMutablePointer<FlutterError?>)
+    -> [SKPaymentTransactionMessage]?
+  {
+
+    let transactions = getPaymentQueueHandler().getUnfinishedTransactions()
+    return transactions.compactMap {
+      FIAObjectTranslator.convertTransaction(toPigeon: $0)
+    }
+  }
+
+  public func storefrontWithError(_ error: AutoreleasingUnsafeMutablePointer<FlutterError?>)
```

Comment:
are these pigeon functions? can you either use `MARK` to denote the section of pigeon functions, or move pigeon protocol conformance to a class extension.